ApiBundle/Tests/Functional/Controller/AuthorizationControllersTest.php

```diff
@@ -43,9 +43,21 @@ public function testTokenCreationAndUsage()
         $this->assertSame('application/json', $this->client->getResponse()->headers->get('content-type'));
         $tokenReponse = json_decode($this->client->getResponse()->getContent(), true);
         $accessToken = $tokenReponse['access_token'];
+        $refreshToken = $tokenReponse['refresh_token'];
 
         $this->client->request('GET', '/api/node/root?access_token=' . $accessToken);
         $this->assertEquals(200, $this->client->getResponse()->getStatusCode());
         $this->assertSame('application/json', $this->client->getResponse()->headers->get('content-type'));
+
+        $this->client->request('GET', '/oauth/access_token?grant_type=refresh_token&refresh_token=' . $refreshToken, array(), array(), array('PHP_AUTH_USER' => 'test_key', 'PHP_AUTH_PW' => 'test_secret'));
+        $this->assertEquals(200, $this->client->getResponse()->getStatusCode());
+        $this->assertSame('application/json', $this->client->getResponse()->headers->get('content-type'));
+        $newTokenReponse = json_decode($this->client->getResponse()->getContent(), true);
+        $newAccessToken = $newTokenReponse['access_token'];
+        $this->assertNotSame($accessToken, $newAccessToken);
+
+        $this->client->request('GET', '/api/node/root?access_token=' . $newAccessToken);
+        $this->assertEquals(200, $this->client->getResponse()->getStatusCode());
+        $this->assertSame('application/json', $this->client->getResponse()->headers->get('content-type'));
     }
```

Comment:
may be add a test with the older token which verifies that it is no longer valid.